src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -24,6 +24,7 @@
 from azure.cli.core.profiles import ResourceType
 
 logger = get_logger(__name__)
+EXCLUDED_NON_CLIENT_PARAMS = list(set(EXCLUDED_PARAMS) - set(['self', 'client']))
```

Comment:
What is this?

Reply:
It's the list of excluded params aside from 'self' and 'client', which are used for client_factory.
'generic_wait' had a bug where if the operation signature included 'client', it would not be used for the client_factory (only 'self' would).
generic_update was doing the right thing: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/arm.py#L442